### PR TITLE
Color refactoring

### PR DESCRIPTION
**Pull request by xarkes**
Refactored color system so there is now a color value (RConsPalette) which computes into a printable value (ansi code, RConsPrintablePalette). Color compression is automatically handled, and there is no information loss when using truecolor theme (check monokai).

### CODE REVIEW

**Comment by xarkes** [comment]
Note that there is some work remaining (like using an enum for color names) which could simplify the code.
Also we shall stop using Color_XXX macros and add or use existing palette colors.

**Comment by xarkes** [comment]
I can't fix AppVeyor. On MSVC it seems that if I use `#define RColor_BLACK (RColor){ 0, 0, 0, 0 }` it will consider it as a non constant in pal.c, line 82 https://github.com/radare/radare2/pull/9464/files#diff-956422fb542ea7cb490dda6e154b37dbR82
However if I remove the cast `(RColor)`, gcc will consider the define as not valid (`error: expected expression before ‘{’ token`)

**Comment by radare** [comment]
nice pr! did u tested on windows?

**Comment by xarkes** [comment]
No I wanted to download the artifact from Appveyor to test it but the build
fails.

File: libr/include/r_cons.h [hunk 1]
```diff
@@ -431,6 +498,35 @@ typedef struct r_cons_t {
 #define Color_BBLUE     "\x1b[1;34m"
 #define Color_BGRAY     "\x1b[1;38m"
 
+#define RColor_NULL      (RColor){ 0x00,         0x00, 0x00, 0x00 }
```
Comment (XVilka) [inline libr/include/r_cons.h]: I think type cast here causes the MSVC faile

**Comment by xarkes** [comment]
Yes that's what I said in my comment above. However if I remove it, gcc
says the define is invalid.

**Comment by XVilka** [comment]
@xarkes for GCC it may require to put define in parentheses, like:
```c
#define RColor_NULL      ({ 0x00, 0x00, 0x00, 0x00 })
```

**Comment by XVilka** [comment]
Nevermind, it will not work too.

**Comment by XVilka** [comment]
Well, just add a dumb
```c
#ifdef _MSC_VER
#define RCAST(x)
#else
#define RCAST(x) (RColor)
#endif
#define RColor_NULL     RCAST() { 0x00,         0x00, 0x00, 0x00 }
```
or something similar

File: libr/cons/cons.c [hunk 2]
```diff
@@ -71,64 +71,35 @@ static inline void r_cons_write(const char *buf, int len) {
...
+	case 8: rcolor = RColor_YELLOW; break;
+	case 9: rcolor = RColor_BYELLOW; break;
+	case 10: rcolor = RColor_CYAN; break;
+	case 11: rcolor = RColor_BCYAN; break;
+	case 12: rcolor = RColor_BLUE; break;
+	case 13: rcolor = RColor_BBLUE; break;
+	case 14: rcolor = RColor_GRAY; break;
+	case 15: rcolor = RColor_BGRAY; break;
+	}
+	return rcolor;
```
Comment (XVilka) [inline libr/cons/cons.c]: Do not return structures allocated in the stack.

**Comment by XVilka** [comment]
So I think now it is good to merge?

**Comment by xarkes** [comment]
Ok, so that is the behaviour on Windows:
image
Works fine with conemu, and as expected windows console does not handle truecolor.

**Pull request merged**